### PR TITLE
Add flag for protecting cdrom devices during populate

### PR DESCRIPTION
**Pull request by rvykydal**
Related: rhbz#1671713

### CODE REVIEW

**Comment by rvykydal** [comment]
Please see the discussion in the related https://github.com/rhinstaller/anaconda/pull/1977

File: blivet/flags.py [hunk 1]
```diff
@@ -77,6 +77,9 @@ def __init__(self):
         # (so far only for LUKS)
         self.discard_new = False
 
+        # wheteher cdroms should be made protected during devicetree population
```
Comment (vpodzime) [inline blivet/flags.py]: *whether*

File: blivet/flags.py [hunk 2]
```diff
@@ -77,6 +77,9 @@ def __init__(self):
...
+        self.protect_cdroms = True
```
Comment (vpodzime) [inline blivet/flags.py]: Any reason why this should be configurable? Let's not introduce flags for absurd/artificial cases.

File: blivet/devicetree.py [hunk 3]
```diff
@@ -936,6 +936,12 @@ def _hide_ignored_disks(self):
                 if ignored:
                     self.hide(disk)
 
+    def _protect_cdroms(self):
+        # protect cdrom devices
+        for device in self._devices:
+            if device.type == "cdrom":
+                log.debug("protecting cdrom %s", device.name)
+                device.protected = True
```
Comment (vpodzime) [inline blivet/devicetree.py]: Shouldn't this be done when the CD-ROMs are being "populated-in"?

File: blivet/flags.py (hunk 2 above)
Comment (dwlehman) [inline blivet/flags.py]: This should default to False.

Comment (rvykydal) [reply blivet/flags.py]: The reason is that to be safe - I wanted to apply the change only to image installations (or better - installations on running system in the future).
This approach is definitely quite ad-hoc solution.
See the discussion in related anaconda PR: https://github.com/rhinstaller/anaconda/pull/1977
Maybe we could set OpticalDevices as protected by default?
I was also thinking the devices could be protected as being read-only (if _mark_readonly_device was not limited to disks but was applied also for cdroms) but apparently cdrom device is not read-only.

File: blivet/devicetree.py (hunk 3 above)
Comment (rvykydal) [reply blivet/devicetree.py]: In the optical device populator run() ? Would it be OK to use the flag then?

File: blivet/flags.py (hunk 2 above)
Comment (rvykydal) [reply blivet/flags.py]: Ah, sure.

**Comment by rvykydal** [comment]
After talking to Vojta I moved application of the flag to OpticalDevice "protected" property.
As for a solution without need for the special flag, I opened https://github.com/storaged-project/blivet/issues/779.

File: blivet/flags.py (hunk 2 above)
Comment (rvykydal) [reply blivet/flags.py]: @dwlehman I fixed the default value and changed the approach a bit (see below)

**Comment by rvykydal** [comment]
Just updated rhbz reference to BZ clone for blivet componet.

**Comment by vojtechtrefny** [comment]
@rvykydal Can we merge this or are we waiting for some changes in Anaconda?

**Comment by dwlehman** [comment]
I still prefer another approach as indicated in discussion of #779. The `teardown_all` at the end of populate only runs when blivet is being driven by anaconda. If it turns out that removing that resolves the issue I would vastly prefer that approach to adding new logic and a flag for such a specific circumstance.

**Comment by poncovka** [comment]
This change can be handled in Anaconda now. See https://github.com/storaged-project/blivet/issues/779#issuecomment-572522589.

**Comment by rvykydal** [comment]
I think we can close this PR - it was added to python-blivet for rhel as downstream patch. I've created jira issue for porting of the BZ to Fedora, building on https://github.com/storaged-project/blivet/pull/780 which has been merged to 3.1 devel.

**Pull request closed without merging**